src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper_windows.cc

```diff
@@ -75,6 +67,14 @@ static bool inner_maybe_resolve_localhost_manually_locked(
     ipv4_loopback_addr.sin_port = numeric_port;
     (*addrs)->emplace_back(&ipv4_loopback_addr, sizeof(ipv4_loopback_addr),
                            nullptr /* args */);
+    // Append the ipv6 loopback address.
```

Comment:
What's the reason for moving this code?  Does the order matter?

Reply:
This was just a test coverage thing. Because tests can assume that the IPv6 loopback is available, but they can't necessarily assume that the IPv4 loopback is available, returning the IPv4 loopback here first allows our tests to verify that the address sorter is actually being used, by checking that the IPv4 loopback was demoted.

That said, this change is not really necessary and is a bit forced. I can revert this diff if we want

Reply:
It does seem a little artificial.  No matter which order we return them in, there is presumably a test case in which they should actually be sorted in the other order, so the order in the code probably doesn't matter much.

Anyway, this doesn't seem to hurt anything, so I'll leave it up to you.

Reply:
The test case in which we expect the ipv4 loopback to come first didn't exist (due to trickiness since environment our tests run in not necessarily having ipv4 but always having ipv6). I've updated this PR to add such a test in though, using a mock, and I've reverted this code change.